### PR TITLE
[Messenger] Document `#[AsMessage]` attribute

### LINKED ISSUE #20002: [Messenger] Introduce `#[AsMessage]` attribute for message routing
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#57507
| PR author(s) | @pounard
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by pounard**
Fixes #20002

Document the newly merged `#[AsMessage]` attribute https://github.com/symfony/symfony/pull/57507

### CODE REVIEW

**Comment by MrYamous** [comment]
I think we should also add it to the [Attribute Overwiew page](https://symfony.com/doc/7.2/reference/attributes.html)

File: messenger.rst [hunk 1]
```diff
@@ -345,6 +345,56 @@ to multiple transports:
...
+            framework:
+                messenger:
+                    routing:
+                        # override class attribute
+                        'App\Message\SmsNotification': sync
+
+.. tip::
+
+    The `$transport` parameter can be either a `string` or an `array`
+    or `string`: configuring multiple transports is possible.
```
Comment (nesl247) [inline messenger.rst]:
```suggestion
    The `$transport` parameter can be either a `string` or an `array`: configuring multiple transports is possible.
```

Unless I'm reading this wrong on the mobile app, string is being repeated.

File: messenger.rst [hunk 2]
```diff
@@ -345,6 +345,56 @@ to multiple transports:
     name as its only argument. For more information about stamps, see
     `Envelopes & Stamps`_.
 
+Configuring Routing Using Attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can optionally use the `#[AsMessage]` attribute to configure message transport:
+
+.. code-block:: php
```
Comment (OskarStark) [inline messenger.rst]:
```suggestion
You can optionally use the `#[AsMessage]` attribute to configure message transport::
```

**Comment by pounard** [comment]
Sorry, I took a few days off. I'll be back in a few days for fixing this.

**Comment by javiereguiluz** [comment]
@pounard friendly ping. This is almost ready, so I hope you can find some time to finish it. Thanks a lot!

**Comment by pounard** [comment]
> @pounard friendly ping. This is almost ready, so I hope you can find some time to finish it. Thanks a lot!

I'm sorry, I spent more time off work than it was initially planned. I will be back in september.

**Comment by pounard** [comment]
@MrYamous

> I think we should also add it to the [Attribute Overwiew page](https://symfony.com/doc/7.2/reference/attributes.html)

I just added it, thanks for pointing this.

**Comment by pounard** [comment]
@javiereguiluz Ready for review once again! Sorry for keeping this waiting.

**Comment by pounard** [comment]
I have really no idea why the "Code Blocks" action fails. Any ideas?

File: messenger.rst [hunk 3]
```diff
@@ -345,6 +345,55 @@ to multiple transports:
...
+                    routing:
+                        # override class attribute
+                        'App\Message\SmsNotification': sync
+
+.. tip::
+
+    The `$transport` parameter can be either a `string` or an `array`: configuring multiple
+    transports is possible. You may also repeat the attribute if you prefer instead of using
+    an array.
+
```
Comment (alamirault) [inline messenger.rst:396]: Can you also add a `.. versionadded:: 7.2` section ?

**Comment by javiereguiluz** [comment]
Pierre, thanks for contributing this feature and its docs.

While merging, I reworded your contribution (See https://github.com/symfony/symfony-docs/commit/d926bbb4e7cb2cd75e0f2f444a3b41dc3a1938d5). It looks like I changed everything ... but that's not true. I reused all your original contents, but merged them into the previous section. I did this because in the rest of the docs we always display the PHP attributes alongside the YAML/XML/PHP config files. Thanks!

File: messenger.rst (hunk 3 above)
Comment (javiereguiluz) [reply messenger.rst:396]: I added this while merging. Thanks.

**Pull request merged**